.github/workflows/pull-request-publish.yml

```diff
@@ -37,6 +37,12 @@ jobs:
           distribution: "zulu"
           java-version: "17"
 
+      - name: Setup Node
```

Comment:
We already use gradle in this project. I'd suggest we should leverage the gradle plugin and actually trigger npm tasks from within the gradle plugin, which would also take care automatically about downloading appropriate node versions. This would be more aligned with what we're doing in the webapp (you can use the `build.gradle` of the webapp as a reference), and makes sure more of our build stays within the build system and doesn't move into the workflow files.

Reply:
+1 to keeping build steps out of the github workflow files

as for using gradle plugins for npm, I am sort of ambivalent, but I do think it makes sense to be able to run the generator using the build system (gradle).  A nice property of the gradle plugin is also that it uses a node/npm setup that is independent of whatever your local environment is using. 

If you're curious, for the python generator, there is a gradle task for it here https://github.com/airbytehq/airbyte-protocol/blob/2f27a76d070bf8a24b6ebd3873208339ed5ed475/protocol-models/build.gradle#L28-L34
but it simply calls a wrapper script that runs the actual generator script in a docker container.  So in this way, we get ability to run the generator from gradle, we can run the generator without mutating our local dev env, and we can even run the generator standalone (independent of github workflows or gradle).  So that might be an option too.